src/main/java/de/hpi/bpt/argos/util/threading/BackgroundWorkerPoolImpl.java

```diff
@@ -0,0 +1,59 @@
+package de.hpi.bpt.argos.util.threading;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+
+/**
+ * {@inheritDoc}
+ * This is the implementation.
+ */
+public class BackgroundWorkerPoolImpl<DataType> implements BackgroundWorkerPool<DataType> {
+
+	private List<BackgroundWorker<DataType>> workers;
+	private int nextWorker;
+
+	/**
+	 * This constructor initializes all members with their default values.
+	 */
+	public BackgroundWorkerPoolImpl() {
+		workers = new ArrayList<>();
+		nextWorker = 0;
+	}
+
+	/**
+	 * This method creates a new backgroundWorkerPool.
+	 * @param workers - the amount of workers in the pool
+	 * @param action - the action, which should be performed by the workers
+	 * @param <DataType> - the dataType of the data objects, which will be processed by the workers
+	 * @return - the new backgroundWorkerPool
+	 */
+	public static <DataType> BackgroundWorkerPool<DataType> Create(int workers, Consumer<DataType> action) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS00100)